pkg/cloud/aws/actuators/machine/actuator_test.go

```diff
@@ -292,3 +292,43 @@ func mockTerminateInstances(mockAWSClient *mockaws.MockClient) {
 	mockAWSClient.EXPECT().TerminateInstances(gomock.Any()).Return(
 		&ec2.TerminateInstancesOutput{}, nil)
 }
+
+func TestRemoveDuplicatedTags(t *testing.T) {
+	cases := []struct {
+		tagList          []*ec2.Tag
+		deduplicatedList []*ec2.Tag
+	}{
+		{
+			// no duplicate tags
+			tagList: []*ec2.Tag{
+				{Key: aws.String("clusterID"), Value: aws.String("test-ClusterIDValue")},
+				{Key: aws.String("clusterName"), Value: aws.String("test-ClusterNameValue")},
+			},
+			deduplicatedList: []*ec2.Tag{
+				{Key: aws.String("clusterID"), Value: aws.String("test-ClusterIDValue")},
+				{Key: aws.String("clusterName"), Value: aws.String("test-ClusterNameValue")},
+			},
+		},
+		{
+			// multiple duplicate tags
+			tagList: []*ec2.Tag{
+				{Key: aws.String("clusterID"), Value: aws.String("test-ClusterIDValue")},
+				{Key: aws.String("clusterName"), Value: aws.String("test-ClusterNameValue")},
+				{Key: aws.String("clusterSize"), Value: aws.String("test-ClusterSizeValue")},
+				{Key: aws.String("clusterName"), Value: aws.String("test-ClusterNameDuplicatedValue")},
+				{Key: aws.String("clusterSize"), Value: aws.String("test-ClusterSizeDuplicatedValue")},
+			},
+			deduplicatedList: []*ec2.Tag{
+				{Key: aws.String("clusterID"), Value: aws.String("test-ClusterIDValue")},
+				{Key: aws.String("clusterName"), Value: aws.String("test-ClusterNameValue")},
+				{Key: aws.String("clusterSize"), Value: aws.String("test-ClusterSizeValue")},
+			},
+		},
+	}
+
+	for i, c := range cases {
+		if len(c.deduplicatedList) != len(removeDuplicatedTags(c.tagList)) {
```

Comment:
The length doesn't guarantee correctness. Better to have `expected` and `actual` and use `reflect.DeepEquals` to verify the result.